Sprint 43 release notes (#153)
validmind/documentation: site/releases/2024-feb-14/highlights.qmd

---
title: "February 14, 2024"
---

## Release highlights

### ValidMind Developer Framework (v1.26.6)

<!---[SC-2706] Documentation templates should allow rendering more than one unique metric result for the same metric by @AnilSorathiya in [#322](https://github.com/validmind/developer-framework/pull/322)--->

> #### Documentation templates now support multiple unique results for the same metric

Reviewer: ```suggestion
#### Documentation templates now tracking each metric result with a unique identifier
```

Reviewer 2: ^ The suggestion is missing a connecting verb. "now tracking" should be "now supports tracking"?

Author: Changed to `Documentation templates now supports tracking each metric result with a unique identifier`, as suggested by @cachafla.

> Documentation templates have been updated to support logging multiple unique results for the same test which enables you to run the same test across different datasets or models. To make use of this new feature, you add a unique `result_id` identifier as a suffix to a `content_id` identifier in the content block definition of a `metric` or `test` content type. 

Reviewer: ```suggestion
Documentation templates have been updated to support logging each metric run as a unique result, making it possible to run same test across different datasets or models. To make use of this new feature, you add a unique `result_id` identifier as a suffix to a `content_id` identifier in the content block definition of a `metric` or `test` content type. 
```

```yaml
- content_type: test
  content_id: validmind.data_validation.Skewness:training_data
- content_type: metric
  content_id: validmind.data_validation.Skewness:test_data
```

You can configure each of these unique `content_id` identifiers by passing the appropriate `config` and `inputs` in `run_documentation_tests()` or `run_test()`. For example, to configure two separate tests for `Skewness` using different datasets and parameters:

```python
test = vm.tests.run_test(
    test_id="validmind.data_validation.Skewness:training_data",
    params={
        "max_threshold": 1
    },
    dataset=vm_train_ds,
)
test.log()

test = vm.tests.run_test(
    test_id="validmind.data_validation.Skewness:test_data",
    params={
        "max_threshold": 1.5
    },
    dataset=vm_test_ds
)
test.log()
```

Try it yourself: [Rendering more than one unique result for the same metric](../../notebooks/how_to/document_multiple_results_for_the_same_test.ipynb)

<!---[SC 3073] `run_documentation_tests()` should allow specifying inputs for any test individually by @AnilSorathiya in [#327](https://github.com/validmind/developer-framework/pull/327)--->

> #### Specify inputs for tests individually with `run_documentation_tests()`

Reviewer: ```suggestion
#### Easier specification of inputs for individual tests with `run_documentation_tests()`
```

```python
full_suite = vm.run_documentation_tests(
    inputs = {
        ...
    },
    config={
        "validmind.data_validation.SomeTest": {
            "min_threshold": 1
        }
    }
)
```

With the updated function, `config` can now pass both `params` and `inputs` to each declared test. For example, to specify what model should be passed to each individual test instance:

```python
full_suite = vm.run_documentation_tests(
    inputs = {
        "dataset": vm_dataset,
        "model": xgb_model
    },
    config = {
        "validmind..model_validation.Accuracy:xgb_model": {
            "params": { threshold: 0.5 },
            "inputs": { "model": xgb_model }
        },
        "validmind..model_validation.Accuracy:lr_model": {
            "params": { threshold: 0.3 },
            "inputs": { "model": lr_model }
        },
    }
)
```

Here, the top-level `inputs` parameter acts as a global `inputs` parameter, and the individual tests can customize what they see as the input model via their own `config` parameters.

<!---[SC-2330] Ability to see available task types and tags by @juanmleng in [#317](https://github.com/validmind/developer-framework/pull/317)--->

> #### See available task types and tags

Reviewer: ```suggestion
#### View available task types and tags to filter tests
```

To enable model developers to know what task types and tags are available to filter on, we have made some updates to our developer framework:

- New `list_task_types()` and `list_tags()` endpoints enable you to list all available `task_type` and `tags` across all test classes
- New `list_tasks_and_tags()` endpoint enable you to list which `tags` are associated to which `task_type`

<img src="28250b4a-fcc7-4863-9a47-55d73698a5d4.png" alt="New indicator that specifies the user who made updates through the developer framework" style="width: 80%; border: 1px solid #333; box-shadow: 5px 5px 5px #ccc, -5px 5px 5px #ccc; border-radius: 5px;">

Try it: [Exploring Tests in the Developer Framework](../../notebooks/how_to/explore_tests.ipynb)

<!---John6797/sc 2943/developer framework should register documentation by @johnwalz97 in [#324](https://github.com/validmind/developer-framework/pull/324)--->

> #### Developer framework documentation inputs

Reviewer: ```suggestion
#### Developer framework documentation inputs tracking
```

We have added a new feature that tracks which datasets and models are used when running tests. Now, when you initialize datasets or models with `vm.init_dataset()` and `vm.init_model()`, we link those inputs with the test results they generate. This makes it clear which inputs were used for each result, improving transparency and making it easier to understand test outcomes. This update does not require any changes to your code and works with existing `init` methods.

### ValidMind Platform UI (v1.13.13)

<!---Update events to show users who ran the developer framework actions by @cachafla in [#532](https://github.com/validmind/frontend/pull/532)--->

> #### Update events to show users who ran the developer framework actions

Reviewer: ```suggestion
#### Update events history to show which users ran actions on the developer framework
```

Update rendering of events related to the developer framework. We are now showing the user who ran the action instead of a generic "Developer Framework" actor name

::: {.grid}
::: {.g-col-6}
**Before**

<img src="78586ce2-4e9d-4613-9738-b09db49c2d5e.png" alt="Old indicator that says that the developer framework logged an event" style="width: 100%; border: 1px solid #333; box-shadow: 5px 5px 5px #ccc, -5px 5px 5px #ccc; border-radius: 5px;">

:::

::: {.g-col-6}
**After**

<img src="8b643ea9-61c6-445e-bf2a-422d94193ffc.png" alt="New indicator that specifies the user who made updates through the developer framework" style="width: 100%; border: 1px solid #333; box-shadow: 5px 5px 5px #ccc, -5px 5px 5px #ccc; border-radius: 5px;">
:::
:::

<!---Nrichers/sc 2163/revisit once again the getting started page by @nrichers in [#543](https://github.com/validmind/frontend/pull/543)--->

> #### Simplified instructions for developers

Reviewer: ```suggestion
#### Simplified Getting Started instructions for developers
```

> We simplified the instructions for getting started with the ValidMind Developer Framework in the Platform UI. These instructions tell you how to use the code snippet for your documentation project with your own model or with one of our code samples.

Reviewer: ```suggestion
We simplified the instructions for getting started with the ValidMind Developer Framework in the Platform UI. These instructions tell you how to use the provided code snippet to connect your documentation project with your own local model development environment (or with one of our code samples).
```

<img src="bf9b42d3-c74f-41f9-97b5-52911e33eb0d.png" alt="New indicator that specifies the user who made updates through the developer framework" style="width: 80%; border: 1px solid #333; box-shadow: 5px 5px 5px #ccc, -5px 5px 5px #ccc; border-radius: 5px;">

## Enhancements

<!---Panchicore/sc 2799/model owner should be able to edit model by @panchicore in [#562](https://github.com/validmind/frontend/pull/562)--->
Model owner can edit model details page
: Model owners can now edit the values for fields displayed on the model details page. Previously it was only possible to edit custom fields defined by your organization.

<!---Reducing load time by using localstorage for Auth0 by @even-steven in [#570](https://github.com/validmind/frontend/pull/570)--->

> Reduced load times for the ValidMind Platform UI

Reviewer: ```suggestion
Enhanced page load times for the ValidMind Platform UI
```

Reviewer 2: Minor comment: my brain interprets "enhanced load times" in a weird way since load time is a measure of quantity. An "enhanced load time" gives me a confusing measure of quantity 🤔.

Author: Yeah, I'm not 100% that we've _enhanced_ load times. I might just change this to `Improved performance for the ...`.

> : We reduced the loading times of our Platform UI to improve the user experience.

Reviewer: ```suggestion
: We made improvements to page load times on our Platform UI to improve the user experience.
```

<!---Added support for filtering model inventory by developers / validators by @even-steven in [#563](https://github.com/validmind/frontend/pull/563)--->
Added support for filtering model inventory by developers / validators

> : Enhanced the Model Inventory filtering to allow filtering by Developers and Validators.

Reviewer: ```suggestion
: Enhanced navigation of the Model Inventory by enabling filtering based on Developers and Validators involved with each model.
```

<!---Support for custom model fields by @panchicore in [#509](https://github.com/validmind/frontend/pull/509)--->
Support for custom model fields in the model inventory

> : The model inventory has been updated to allow organizations to add custom fields. This enhancement enables administrators to customize the model inventory according to your specific organizational needs.

Reviewer: ```suggestion
: The model inventory has been updated to allow organizations to add custom fields. This enhancement enables administrators to customize the model inventory data schema according to your specific organizational needs. This can be done by accessing **Custom Fields** in the **Settings** page.
```

- Multiple Select
- Checkbox
- Number
- URL
- Date
- Date Time
- Email
- Linked Record to User

<!---Adds toggle to see only mentions in Recent Activity > Comments by @gtagle in [#525](https://github.com/validmind/frontend/pull/525)--->
Filter for mentions in recent activity comments
: We implemented a toggle feature in the **Recent Activity** > **Comments** section to filter and display only specific mentions. By default, all comments where the logged-in user has been tagged are displayed by this filter. 

<!---Add CKEditor support to finding description and remediation plan - Introduce the `RichTextContentEditor` component by @panchicore in [#542](https://github.com/validmind/frontend/pull/542)--->
Expanded rich-text editor support
: Forms in the **Findings** and **Validation Report** sections now support the rich-text editor interface. This support enables you to use the editor for your finding descriptions and remediation plans, for example. 

## Bug fixes

<!---Invalid content blocks create errors in run documentation by @cachafla in [#326](https://github.com/validmind/developer-framework/pull/326)--->
Invalid content blocks create errors for `run_documentation_tests()`

> : Fixed an issue where using an invalid test identifier would prevent `run_documentation_tests()` from running all available tests. The full test suite now runs as expected, even when an invalid test identifier is specified.

Reviewer: @cachafla  - does the framework throw some message to the user when an invalid test identifier is specified? If yes, I would clarify here

Reviewer 2: Yes. It throws an error saying it could not find the test specified in the template.
